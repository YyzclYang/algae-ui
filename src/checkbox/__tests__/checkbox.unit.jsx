import React from 'react';
import * as renderer from 'react-test-renderer';
import Checkbox from '../index';
import { mount } from 'enzyme';

describe('Checkbox', () => {
  it('渲染一个 Checkbox', () => {
    const component = renderer.create(<Checkbox>radio</Checkbox>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('接受 checked', () => {
    const component = renderer
      .create(<Checkbox checked>radio</Checkbox>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('接受 indeterminate', () => {
    const component = renderer
      .create(<Checkbox indeterminate>radio</Checkbox>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('接受 disabled', () => {
    const component = renderer
      .create(<Checkbox disabled>radio</Checkbox>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('不传递子节点', () => {
    const component = renderer.create(<Checkbox disabled />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('响应 onChange 事件', () => {
    const fn = jest.fn();
    const component = mount(<Checkbox onChange={fn}>radio</Checkbox>);
    component.find('input').simulate('change');
    expect(fn).toBeCalled();
  });

  it('disabled 时不可响应 onChange 事件', () => {
    const fn = jest.fn();
    const component = mount(
      <Checkbox disabled onChange={fn}>
        radio
      </Checkbox>
    );
    component.find('input').simulate('change');
    expect(fn).not.toBeCalled();
  });
});
