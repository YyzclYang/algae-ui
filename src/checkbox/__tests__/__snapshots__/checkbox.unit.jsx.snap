// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox 不传递子节点 1`] = `
<label
  className="algae-ui-checkbox-wrapper algae-ui-checkbox-disabled"
>
  <span
    className="algae-ui-checkbox"
  >
    <input
      checked={false}
      className="algae-ui-checkbox-input"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
  </span>
</label>
`;

exports[`Checkbox 接受 checked 1`] = `
<label
  className="algae-ui-checkbox-wrapper algae-ui-checkbox-checked"
>
  <span
    className="algae-ui-checkbox"
  >
    <input
      checked={true}
      className="algae-ui-checkbox-input"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span
    className="algae-ui-checkbox-text"
  >
    radio
  </span>
</label>
`;

exports[`Checkbox 接受 disabled 1`] = `
<label
  className="algae-ui-checkbox-wrapper algae-ui-checkbox-disabled"
>
  <span
    className="algae-ui-checkbox"
  >
    <input
      checked={false}
      className="algae-ui-checkbox-input"
      disabled={true}
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span
    className="algae-ui-checkbox-text"
  >
    radio
  </span>
</label>
`;

exports[`Checkbox 接受 indeterminate 1`] = `
<label
  className="algae-ui-checkbox-wrapper algae-ui-checkbox-indeterminate"
>
  <span
    className="algae-ui-checkbox"
  >
    <input
      checked={false}
      className="algae-ui-checkbox-input"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span
    className="algae-ui-checkbox-text"
  >
    radio
  </span>
</label>
`;

exports[`Checkbox 渲染一个 Checkbox 1`] = `
<label
  className="algae-ui-checkbox-wrapper"
>
  <span
    className="algae-ui-checkbox"
  >
    <input
      checked={false}
      className="algae-ui-checkbox-input"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span
    className="algae-ui-checkbox-text"
  >
    radio
  </span>
</label>
`;
